etest_test/profile_test.py

```diff
@@ -0,0 +1,221 @@
+"""Tests for etest.profile."""
+
+import logging
+from typing import Any
+
+import etest.profile as sut
+
+logger = logging.getLogger(__name__)
+
+
+def test_base_arch() -> None:
+    """Ensure base arches work."""
+    for arch in ["amd64", "x86", "arm64"]:
+        result = sut.Profile(logger, False, arch, "glibc", False, True, False)
+
+        # Check profile strings
+        assert result.profile == arch
+        assert result.docker == arch
+
+        # Check arch values
+        assert result.arch == arch
+        assert result.pkg_arch == arch
+        assert result.docker_arch == arch
+
+        # Check profile options
+        assert result.hardened is False
+        assert result.multilib is True
+        assert result.systemd is False
+
+        # Check libc
+        assert result.libc == "glibc"
+
+
+def test_base_ppc64() -> None:
+    """Ensure the ppc64 base arch works."""
+    arch = "ppc64"
+    docker_arch = "ppc64le"
+
+    result = sut.Profile(logger, False, arch, "glibc", False, True, False)
+
+    # Check profile strings
+    assert result.profile == arch
+    assert result.docker == docker_arch
+
+    # Check arch values
+    assert result.arch == arch
+    assert result.pkg_arch == "ppc64"
+    assert result.docker_arch == docker_arch
+
+    # Check profile options
+    assert result.hardened is False
+    assert result.multilib is True
+    assert result.systemd is False
+
+    # Check libc
+    assert result.libc == "glibc"
+
+
+def test_base_arm() -> None:
+    """Ensure ARM base arches work."""
+    docker_arm_mappings = {"armv5": "armv5tel", "armv6": "armv6j_hardfp", "armv7": "armv7a_hardfp"}
+
+    for arch in ["armv5", "armv6", "armv7"]:
+        result = sut.Profile(logger, False, arch, "glibc", False, True, False)
+
+        # Check profile strings
+        assert result.profile == arch
+        assert result.docker == docker_arm_mappings[arch]
+
+        # Check arch values
+        assert result.arch == arch
+        assert result.pkg_arch == "arm"
+        assert result.docker_arch == docker_arm_mappings[arch]
+
+        # Check profile options
+        assert result.hardened is False
+        assert result.multilib is True
+        assert result.systemd is False
+
+        # Check libc
+        assert result.libc == "glibc"
+
+
+def test_libc() -> None:
+    """Ensure alternative libcs work."""
+    arch = "amd64"
+
+    for libc in ["musl", "uclibc"]:
```

Comment:
Check out cartesian products with paremetrise and this becomes even simpler.  In fact, the level of assertions you're doing appear to be checking definitions (which isn't needed).  We can double check this once dataclass and parametrise are in use.